utils/fdb_wrapper_cpp/src/fdbcs.cpp

```diff
@@ -206,6 +223,312 @@ std::shared_ptr<FDBDataBase> DataBaseCreator::createDataBase(const std::string c
   return std::make_shared<FDBDataBase>(database);
 }
 
+Key BoostUIDKeyGenerator::generateKey()
+{
+  return boost::lexical_cast<std::string>(boost::uuids::random_generator()());
+}
+
+uint32_t BoostUIDKeyGenerator::getKeySize()
+{
+  return boost::lexical_cast<std::string>(boost::uuids::random_generator()()).size();
+}
+
+Keys BlobHandler::generateKeys(const uint32_t num)
+{
+  Keys keys;
+  keys.reserve(num);
+  for (uint32_t i = 0; i < num; ++i)
+    keys.push_back(keyGen_->generateKey());
+
+  return keys;
+}
+// FIXME: Put it to util?
+float BlobHandler::log(const uint32_t base, const uint32_t value)
+{
+  return std::log(value) / std::log(base);
+}
+
+void BlobHandler::insertKey(Block& block, const std::string& value)
+{
+  if (!block.first)
+  {
+    block.second.reserve(blockSizeInBytes_);
+    block.second.insert(block.second.end(), keyBlockIdentifier.begin(), keyBlockIdentifier.end());
+    block.first += keyBlockIdentifier.size();
+  }
+  block.second.insert(block.second.begin() + block.first, value.begin(), value.end());
+  block.first += value.size();
+}
+
+size_t BlobHandler::insertData(Block& block, const std::string& blob, const size_t offset)
+{
+  const size_t endOfBlock = std::min(offset + dataBlockSizeInBytes_, blob.size());
+  auto& dataBlock = block.second;
+  if (!block.first)
+  {
+    dataBlock.reserve(blockSizeInBytes_);
+    dataBlock.insert(dataBlock.end(), dataBlockIdentifier.begin(), dataBlockIdentifier.end());
+    block.first += dataBlockIdentifier.size();
+  }
+  dataBlock.insert(dataBlock.begin() + block.first, blob.begin() + offset, blob.begin() + endOfBlock);
```

Comment:
Don't you need to increase block.first by the leng of the data inserted?

Reply:
Not really, the semantics of this function is following - 
1) we have a block to insert with fixed size.
2) we have a blob with fixed size, could be greater than block 
3) we have offset in blob from where we have to start to insert data.

a)If the size of the remaining data in blob is greater than block size, we insert the block until the end.
b)If the size of the remaining data in blob is less than block size, that means that is a last block, because blob is in the end, and we do not need this block for insertion anymore.
Conclusion: so we do not need to increment `block.first` for data block, this is different comparing to the key block.